[recipe] fix: Fix a Typo in One_Step_Off_Policy and Add async of Generative Reward Model in Response Generation

ZhichaoWang970201 opened the pull request:
Fix a typo in verl/workers/fsdp_workers.py: 
    original code: if self.model_config.generation_config is not None
    updated code: if self.generation_config is not None

Add async of generation reward model (GRM):
    As the generative reward model is slow in the call. It is unreasonable to wait for all responses to be generated before sending to GRM for evaluation. So I add an async to start GRM evaluation once individual response generation is finished.

Zhichao-Jimmy970201 committed:
fix a typo of one-step-off-policy and add async of reward generation into the async of response generation so that for each individual response generated, a reward generation will start at once. This will be very useful if the reward model is Generative Reward Model (GRM).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/volcengine/verl?pullRequest=3369) <br/>All committers have signed the CLA.

Zhichao-Jimmy970201 committed:
Apply pre-commit fixes

Zhichao-Jimmy970201 committed:
Update various files and workflows for improved functionality and documentation

- Updated README.md and various requirements files for clarity and accuracy.
- Enhanced setup.py for better installation processes.
- Modified multiple GitHub workflows to improve CI/CD processes and ensure compatibility with recent changes.
- Refined documentation across several sections to provide clearer guidance and examples.
- Adjusted Dockerfiles and scripts for better performance and compatibility with new models and configurations.

This commit aims to streamline the development process and enhance user experience with updated documentation and improved workflows.
